manifests/osfamily/redhat.pp

```diff
@@ -18,12 +18,17 @@
         $platform_and_version = "fedora/${facts['os']['release']['major']}"
       }
       'Amazon': {
-        if ("${facts['os']['release']['major']}" == '2') {
-          $amz_el_version = '7'
-        } else {
-          $amz_el_version = '6'
+        $major_version = $facts['os']['release']['major']
+        $amz_el_version = "${major_version}" ? {
+          '2'     => '7',
+          '2017'  => '6',
```

Comment:
It's possible to update the `system-release` package on AL1:

```
# yum update system-release
Loaded plugins: priorities, update-motd, upgrade-helper
amzn-main                                                                                                        | 2.1 kB  00:00:00     
amzn-updates                                                                                                     | 3.7 kB  00:00:00     
Resolving Dependencies
--> Running transaction check
---> Package system-release.noarch 0:2017.03-0.0 will be updated
---> Package system-release.noarch 0:2018.03-1.3 will be an update
...
# cat /etc/system-release
Amazon Linux AMI release 2018.03
```

In which case `os.release.major` will return `2018`:

```
# rpm -Uvh https://yum.puppet.com/puppet7/el/6/x86_64/puppet-agent-7.27.0-1.el6.x86_64.rpm
Retrieving https://yum.puppet.com/puppet7/el/6/x86_64/puppet-agent-7.27.0-1.el6.x86_64.rpm
warning: /var/tmp/rpm-tmp.tA55H7: Header V4 RSA/SHA512 Signature, key ID 9e61ef26: NOKEY
Preparing...                          ################################# [100%]
Updating / installing...
   1:puppet-agent-7.27.0-1.el6        ################################# [100%]
# /opt/puppetlabs/puppet/bin/facter os.release
{
  full => "2018.03",
  major => "2018",
  minor => "03"
}
```

Maybe just add another branch `'2018' => '6',`?

Reply:
Added condition to handle el/6 system version 2017 & 2028.